functional_tests/scylla_operator/libs/auxiliary.py

```diff
@@ -39,13 +41,20 @@ def update_test_status(self, test_name, status, error=None):
         self.test_data[test_name] = (status, error)
 
     def get_test_failures(self):
-        pass
+        for test_name, test_data in self.test_data.items():
+            if not test_data[0] in ('SUCCESS', 'SKIPPED'):
+                TestFrameworkEvent(
+                    source=self.__class__.__name__,
+                    source_method=test_name,
+                    message=test_data[1],
```

Comment:
It is unsafe coding. Presence of the index is not verified. The same above.

Reply:
Changing it from tuple to a dict, would be safer ?

Reply:
Safer is to avoid sudden `IndexError` and `TypeError`. Simple `len()` check is enough.